### PR TITLE
Support for Context in GetMetricHistory and GetMetricHistories in Python API

### LINKED ISSUE #363: Metric contexts: add new methods to the Python API for metric querying via MLFlow endpoints
- `get_metric_history` and `get_metric_histories` need to support requesting and returning contexts

### PR DESCRIPTION
**Pull request by fabiovincenzi**
Fixes #363 
Added support for context in `get_metric_history` in the new fasttrackml python client, included `get_metric_histories` and adapted it to work with context.

### CODE REVIEW

File: python/fasttrackml/client.py [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+    def get_metric_histories(
+        self,
+        experiment_ids: Optional[List[str]] = None,
+        run_ids: Optional[List[str]] = None,
+        metric_keys: Optional[List[str]] = None,
+        index: str = "step",
+        run_view_type: int = ViewType.ACTIVE_ONLY,
+        max_results: int = 10000000,
+        search_all_experiments: bool = False,
+        experiment_names: Optional[List[str]] = None,
```
Comment (suprjinx) [inline python/fasttrackml/client.py:234]: This endpoint now supports a `context` request param for filtering the results, so we should add an optional dict here.

Comment (fabiovincenzi) [reply python/fasttrackml/client.py:234]: added!

File: docs/fasttrackml_python_client.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
+
+## FastTrackML Python Client Overview
+
+`fasttrackml` is a Python package that fully leverage the capabilities of MLFlow, FastTrackML extends its functionality by providing additional methods for a more convenient and efficient user experience.
```
Comment (suprjinx) [inline docs/fasttrackml_python_client.md:4]:
```
`fasttrackml` is a Python package that incorporates and enhances the capabilities MLFlow's tracking client package. While offerring the same interface as MLFlow, `fasttrackml` adds some additional methods for convenience and better performance with the FastTrackML tracking server.
```

Comment (suprjinx) [reply docs/fasttrackml_python_client.md:4]: suggestion ^^

File: docs/fasttrackml_python_client.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    ```python
+	from fasttrackml import FasttrackmlClient
+
+	# Create a FasttrackmlClient instance
+	client = FasttrackmlClient()
+
+	# Fetch metric histories for multiple runs and metrics
+	run_ids = ["run_id1", "run_id2"]  # Replace with valid run IDs
+	metric_keys = ["metric1", "metric2"]  # Replace with desired metric keys
+	metric_histories_df = client.get_metric_histories(run_ids=run_ids, metric_keys=metric_keys)
```
Comment (suprjinx) [inline docs/fasttrackml_python_client.md:68]: should this show use of `context` param?

**Review by suprjinx — CHANGES_REQUESTED** [review]
added a suggestion for the readme intro.

**Review by suprjinx — APPROVED** [review]

**Pull request merged**